src/include/miopen/magic_div.hpp

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2021 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_MLOPEN_MAGIC_DIV_HPP_
+#define GUARD_MLOPEN_MAGIC_DIV_HPP_
+
+#include <cstdint>
+#include <cassert>
+
+namespace miopen {
+
+struct magic_div_u32
+{
+    uint32_t magic;
+    uint8_t shift;
+};
+
+using magic_div_u32_t = magic_div_u32;
```

Comment:
:red_circle: Something similar resides in src/include/miopen/conv/invokers/impl_gemm_dynamic.hpp. We do NOT create copies of existing code.

Reply:
I understand this, the reason I don't want to use that code in `src/composable_kernel/composable_kernel/include/utility/magic_division.hpp` is that code reside in `ck` namespace. But here I use magic div and seems not related to the ck namespace. I mean, this would be code in common. If this really is a issue, maybe fix this in other PR? It may need to think how to reorganize this code

Reply:
Or course code duplication is an issue. It can kill any project, slowly but inevitably.